src/lib.rs

```diff
@@ -605,6 +608,60 @@ macro_rules! query_group {
                     db_trait($query_trait);
                     query_type($QueryType);
                     key($($key_name: $key_ty),*);
+                    lifetime();
+                ]
+            }
+        )*
+    };
+
+    // Base case with lifetime: found the trait body
+    (
+        attr[$($trait_attr:tt)*];
+        headers[$v:vis, $query_trait:ident, $($header:tt)*];
+        tokens[{
+            $(
+                $(#[$method_attr:meta])*
+                fn $method_name:ident($($key_name:ident: $key_ty:ty),* $(,)*) -> $value_ty:ty {
+                    type $QueryType:ident;
+                    $(storage $storage:tt;)* // FIXME(rust-lang/rust#48075) should be `?`
+                    $(use fn $fn_path:path;)* // FIXME(rust-lang/rust#48075) should be `?`
+                }
+            )*
+        }];
+        lifetime[$lt:lifetime];
+    ) => {
+        $($trait_attr)* $v trait $query_trait<$lt>: $($crate::plumbing::GetQueryTable<$QueryType<$lt>> +)* $($header)* {
```

Comment:
Do we really need a separate arm for this? Can't we change this to

```
trait $query_trait<$($lt),*> 
```

so that it works whether or not a lifetime was provided?